src/io/mwaf.rs

```diff
@@ -171,50 +178,85 @@ impl FlagFileSet {
         let first_timestep = timestep_range.start;
         let gps_start = corr_ctx.timesteps[first_timestep].gps_time_ms as f64 / 1e3
             + corr_ctx.metafits_context.corr_int_time_ms as f64 / 2e3;
-        let num_rows = timestep_range.len() * corr_ctx.metafits_context.num_baselines;
+
+        let ant_pairs = vis_sel.get_ant_pairs(&corr_ctx.metafits_context);
+        let mut autos_included = false;
+        for &(a, b) in &ant_pairs {
+            if a == b {
+                autos_included = true;
+                break;
+            }
+        }
+
+        let unique_ant_indices = ant_pairs
+            .into_iter()
+            .flat_map(|(a, b)| [a, b].into_iter())
+            .collect::<std::collections::HashSet<_>>();
+        let mut unique_ant_indices_sorted = unique_ant_indices.into_iter().collect::<Vec<_>>();
+        unique_ant_indices_sorted.sort_unstable();
+
+        let num_ants = unique_ant_indices_sorted.len();
+        let mut num_baselines = (num_ants * (num_ants + 1)) / 2;
```

Comment:
this isn't always true, I'd like to be able to select arbitrary baselines

Reply:
although there are other places in the code like line 669 that use the same logic, so this is fine for now.